build.assets/windows/build.ps1

```diff
@@ -288,18 +288,32 @@ function Invoke-SignBinary {
     <#
     .SYNOPSIS
     Signs the provided binary with the base64-encoded certificate listed in "$WINDOWS_SIGNING_CERT"
+    .PARAMETER UnsignedBinaryPath
+    The path to the unsigned binary.
+    .PARAMETER SignedBinaryPath
+    The path where the signed binary should be written. If not provided, then the signed binary will
+    be written to a temporary path, and then moved to the unsigned binary path.
     #>
     [CmdletBinding()]
     param(
         [Parameter(Mandatory)]
         [string] $UnsignedBinaryPath,
 
-        [Parameter(Mandatory)]
+        [Parameter()]
         [string] $SignedBinaryPath
     )
 
+    if (! $SignedBinaryPath) {
+        $ShouldMoveSignedBinary = true
+        $SignedBinaryPath = Join-Path -Path $(New-TempDirectory) -ChildPath "signed.exe"
+    }
+
     Write-Host "Signing $UnsignedBinaryPath using WSL sign-binary script:"
     wsl-ubuntu-command sign-binary "$UnsignedBinaryPath" "$SignedBinaryPath"
+
+    if ($ShouldMoveSignedBinary) {
```

Comment:
This var is uninitialised if `$SignedBinaryPath` is provided as a parameter. Does this matter? Should it be initialised to `false` before setting it to true conditionally above?

Reply:
No it's fine in powershell. Undefined values are considered null, which evaluates to false. Here's an example:

```powershell
Write-Host $SomeInvalidVar # Nothing is logged

if ($SomeMadeUpVar) {
  Write-Host "If this is logged then the PR logic is flawed"
} else {
  Write-Host "Undefined values are null, which is falsey" # This is printed
}
```